### PR TITLE
Migrate attributes defined config to new scarb plugins

### LINKED ISSUE #1905: Migrate attributes defined config to new scarb plugins
### Which components does the task require to be changed? (think hard pls)

snforge

### Description

We should try migrating this feature into the new scarb plugin feature, so that we don't have to maintain it in the scarb codebase.

### PR DESCRIPTION
**Pull request by Draggu**
Closes #1905

## Introduced changes

- added scarb plugins

Original stack:
- https://github.com/foundry-rs/starknet-foundry/pull/2044
- https://github.com/foundry-rs/starknet-foundry/pull/2191
- https://github.com/foundry-rs/starknet-foundry/pull/2237
- https://github.com/foundry-rs/starknet-foundry/pull/2250
- https://github.com/foundry-rs/starknet-foundry/pull/2264
- https://github.com/foundry-rs/starknet-foundry/pull/2350
- https://github.com/foundry-rs/starknet-foundry/pull/2325

### CODE REVIEW

**Review by Arcticae — APPROVED** [review]

File: crates/forge-runner/src/build_trace_data.rs [hunk 1]
```diff
@@ -148,7 +144,11 @@ pub fn build_profiler_execution_resources(
         vm_resources: VmExecutionResources {
             n_steps: execution_resources.n_steps,
             n_memory_holes: execution_resources.n_memory_holes,
-            builtin_instance_counter: execution_resources.builtin_instance_counter,
+            builtin_instance_counter: execution_resources
+                .builtin_instance_counter
+                .into_iter()
+                .map(|(key, value)| (key.to_str_with_suffix().to_owned(), value))
+                .collect(),
```
Comment (cptartur) [inline crates/forge-runner/src/build_trace_data.rs:151]: Why this change?

File: crates/forge-runner/src/gas.rs [hunk 2]
```diff
@@ -123,14 +126,46 @@ fn get_messages_costs(
     starknet_gas_usage + sharp_gas_usage
 }
 
+fn clear_compiled_class_hash_update(
+    state_changes: blockifier::state::cached_state::StateChanges,
+) -> blockifier::state::cached_state::StateChanges {
+    // copy-pasted blockifier::state::cached_state::StateChanges
```
Comment (cptartur) [inline crates/forge-runner/src/gas.rs:132]: Can we link relevant files on gh?

File: crates/forge-runner/src/gas.rs [hunk 3]
```diff
@@ -123,14 +126,46 @@ fn get_messages_costs(
...
+    unsafe {
+        // compiled_class_hash_updates is used only for keeping track of declares
+        // which we don't want to include in gas cost
+        // clear on public struct
+        (*u.public).compiled_class_hashes.clear();
+
+        // cast back into original blockifier::state::cached_state::StateChanges
+        std::mem::ManuallyDrop::<blockifier::state::cached_state::StateChanges>::into_inner(
+            u.origin,
+        )
```
Comment (cptartur) [inline crates/forge-runner/src/gas.rs:159]: Are these comments from the original code?

File: crates/forge-runner/src/running/syscall_handler.rs [hunk 4]
```diff
@@ -25,9 +25,9 @@ pub fn build_syscall_handler<'a>(
         .iter()
         .any(|(ty, _)| ty == &SegmentArenaType::ID)
     {
-        12
+        16
     } else {
-        10
+        14
```
Comment (cptartur) [inline crates/forge-runner/src/running/syscall_handler.rs:30]: How do we know it should be these now?

File: crates/forge-runner/src/running/with_config.rs [hunk 5]
```diff
@@ -50,11 +50,10 @@ pub fn test_target_with_config(test_target_raw: TestTargetRaw) -> Result<TestTar
     .ok_or_else(|| anyhow!("can not get type size map"))?;
 
     let default_executables = vec![];
-    let executables = test_target_raw
-        .sierra_program
-        .debug_info
-        .executables
-        .get("snforge_internal_test_executable")
+    let deubg_info = test_target_raw.sierra_program.debug_info.clone();
```
Comment (cptartur) [inline crates/forge-runner/src/running/with_config.rs:53]: Typo, should be `debug_info`.

File: crates/forge/tests/e2e/running.rs [hunk 6]
```diff
@@ -751,84 +751,84 @@ fn should_panic() {
         output,
         indoc! { r"
         Collected 14 test(s) from should_panic_test package
+        Running 0 test(s) from src/
```
Comment (cptartur) [inline crates/forge/tests/e2e/running.rs:754]: Why this behavior changed?

File: crates/forge-runner/src/running/syscall_handler.rs (hunk 4 above)
Comment (ksew1) [reply crates/forge-runner/src/running/syscall_handler.rs:30]: cc @Draggu

File: crates/forge/tests/e2e/running.rs (hunk 6 above)
Comment (ksew1) [reply crates/forge/tests/e2e/running.rs:754]: It should always be that way. I don't now why it was passing before

File: crates/forge-runner/src/gas.rs (hunk 3 above)
Comment (ksew1) [reply crates/forge-runner/src/gas.rs:159]: Orginal

File: crates/forge-runner/src/build_trace_data.rs (hunk 1 above)
Comment (ksew1) [reply crates/forge-runner/src/build_trace_data.rs:151]: `.collect()` change form `.collect_vec()`. Better type inference

**Review by cptartur — APPROVED** [review]
